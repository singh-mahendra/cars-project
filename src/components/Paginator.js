import React from 'react'
import PropTypes from 'prop-types';

const Paginator = (props) =>{
    
    return (
        <>
        <input type="Button" value="First" onClick={props.navigateToPage(1)}></input>
        <input type="Button" value="Previous" onClick={props.navigateToPage(props.currentPage - 1)}></input>
        <span>Page {props.currentPage} of {props.totalPages}</span>
        <input type="Button" value="Next" onClick={props.navigateToPage(props.currentPage + 1)}></input>
        <input type="Button" value="Last" onClick={props.navigateToPage(props.totalPages)}></input>
        </>
    );
}

Paginator.propTypes = {
    pageNumber: PropTypes.number,
    totalPages: PropTypes.number,
    navigateToPage: PropTypes.func
}

export default Paginator;