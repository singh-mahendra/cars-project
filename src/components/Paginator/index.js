import React from 'react'
import PropTypes from 'prop-types';

const Paginator = (props) =>{
    
    return (
        <section className="paginator">
            <a href="#" onClick={props.navigateToPage(1)}>First</a>
            <a href="#" onClick={props.navigateToPage(props.currentPage - 1)}>Previous</a>
            <span>Page {props.currentPage} of {props.totalPages}</span>
            <a href="#" onClick={props.navigateToPage(props.currentPage + 1)}>Next</a>
            <a href="#" onClick={props.navigateToPage(props.totalPages)}>Last</a>
        </section>
    );
}

Paginator.propTypes = {
    pageNumber: PropTypes.number,
    totalPages: PropTypes.number,
    navigateToPage: PropTypes.func
}

export default Paginator;