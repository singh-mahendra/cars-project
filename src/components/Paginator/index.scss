.paginator{
    a{
      margin: 12px;
      padding: 12px;
    }
  }