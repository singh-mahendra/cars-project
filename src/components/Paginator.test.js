import React from 'react';
import { shallow } from 'enzyme';
import Paginator from './Paginator';

describe("Paginator", () => {
  it("should render my component", () => {
    const pageNumber = 1;
    const navigateToPage = () => {}
    const wrapper = shallow(<Paginator pageNumber={pageNumber} totalPages={pageNumber} navigateToPage={navigateToPage}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});