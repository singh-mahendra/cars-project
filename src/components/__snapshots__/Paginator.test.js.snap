// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should render my component 1`] = `
Array [
  <section
    className="paginator"
  >
    <a
      href="#"
    >
      First
    </a>
    <a
      href="#"
    >
      Previous
    </a>
    <span>
      Page 
       of 
      1
    </span>
    <a
      href="#"
    >
      Next
    </a>
    <a
      href="#"
    >
      Last
    </a>
  </section>,
]
`;
